components/org.wso2.carbon.identity.data.publisher.audit.idp.properties/src/main/java/org/wso2/carbon/identity/data/publisher/audit/idp/properties/impl/ResidentIdPPropertiesDataPublisher.java

```diff
@@ -0,0 +1,155 @@
+                        + "@" + actionHolderTenantDomain);
+            }
+        } catch (UserStoreException e) {
+            log.error("Failed to fetch action holder user store domain for user " + actionHolder);
+        }
+
+        idPProperties.addParameter(AuditDataPublisherConstants.PUBLISHING_TENANT_DOMAINS,
+                AuditDataPublisherUtils.getTenantDomains(actionHolderTenantDomain, tenantDomain));
+
+        publishResidentIdPData(idPProperties);
+        return true;
+    }
+
+    /**
+     * Publish idp related data to IS Analytics.
+     *
+     * @param idPProperties The IdP properties to be published.
+     */
+    private void publishResidentIdPData(IdPProperties idPProperties) {
+        Object[] payloadData = new Object[5];
+        payloadData[0] = idPProperties.getIdPName();
+        payloadData[1] = idPProperties.getTenantDomain();
+        payloadData[2] = new Gson().toJson(idPProperties.getProperties());
+        payloadData[3] = idPProperties.getActionHolder();
+        payloadData[4] = idPProperties.getTimestamp();
+
+        String[] publishingDomains =
+                (String[]) idPProperties.getParameter(AuditDataPublisherConstants.PUBLISHING_TENANT_DOMAINS);
+        if (publishingDomains != null && publishingDomains.length > 0) {
+            try {
+                FrameworkUtils.startTenantFlow(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+                for (String publishingDomain : publishingDomains) {
+                    Object[] metadataArray = AuditDataPublisherUtils.getMetaDataArray(publishingDomain);
+                    org.wso2.carbon.databridge.commons.Event event = new org.wso2.carbon.databridge.commons.Event
+                            (AuditDataPublisherConstants.IDP_PROPERTIES_UPDATE_EVENT_STREAM_NAME, System
+                                    .currentTimeMillis(), metadataArray, null, payloadData);
+                    IdPPropertiesDataPublisherDataHolder.getInstance().getPublisherService().publish(event);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Sending out event : " + event.toString());
+                    }
+                }
+            } finally {
+                FrameworkUtils.endTenantFlow();
+            }
+        }
+    }
+
+    @Override
+    public boolean isEnable() {
+        // Extended to change the default behaviour to not publish data is the config is missing
+        IdentityEventListenerConfig identityEventListenerConfig = IdentityUtil.readEventListenerProperty(
+                IdentityProviderMgtListener.class.getName(), this.getClass().getName());
+        return identityEventListenerConfig != null
+                && StringUtils.isNotBlank(identityEventListenerConfig.getEnable())
+                && Boolean.parseBoolean(identityEventListenerConfig.getEnable());
+    }
+
+    @Override
+    public int getDefaultOrderId() {
+        return AuditDataPublisherConstants.IDP_MGT_LISTENER_DEFAULT_ORDER_ID;
+    }
+}
```

Comment:
Make sure there is a new line after each method signature. Configure your IDE to add that.

Reply:
Added.